x-pack/qa/password-protected-keystore/src/test/java/org/elasticsearch/password_protected_keystore/ReloadSecureSettingsWithPasswordProtectedKeystoreRestIT.java

```diff
@@ -78,10 +78,16 @@ public void testReloadSecureSettingsWithEmptyPassword() throws Exception {
             assertThat(entry.getValue(), instanceOf(Map.class));
             final Map<String, Object> node = (Map<String, Object>) entry.getValue();
             assertThat(node.get("reload_exception"), instanceOf(Map.class));
-            assertThat(ObjectPath.eval("reload_exception.reason", node), anyOf(
-                equalTo("Provided keystore password was incorrect"),
-                equalTo("Keystore has been corrupted or tampered with")));
-            assertThat(ObjectPath.eval("reload_exception.type", node), equalTo("security_exception"));
+            if (inFipsJvm()) {
+                assertThat(ObjectPath.eval("reload_exception.reason", node), containsString(
+                    "Error generating an encryption key from the provided password"));
+                assertThat(ObjectPath.eval("reload_exception.type", node), equalTo("general_security_exception"));
+            } else {
+                assertThat(ObjectPath.eval("reload_exception.reason", node), anyOf(
+                    equalTo("Provided keystore password was incorrect"),
+                    equalTo("Keystore has been corrupted or tampered with")));
+                assertThat(ObjectPath.eval("reload_exception.type", node), equalTo("security_exception"));
+            }
```

Comment:
I'm not convinced that we actually care about FIPS vs non-FIPS here do we?
Why wouldn't we just have:
```suggestion
            assertThat(ObjectPath.eval("reload_exception.reason", node), anyOf(
                equalTo("Provided keystore password was incorrect"),
                equalTo("Keystore has been corrupted or tampered with"),
                containsString("Error generating an encryption key from the provided password") // FIPS
            ));
            assertThat(ObjectPath.eval("reload_exception.type", node), 
                // Depends on exact security provider (eg Sun vs BCFIPS)
                anyOf(equalTo("security_exception"), equalTo("general_security_exception"))
            );
```